drivers/can/can_renesas_ra.c

```diff
@@ -0,0 +1,1171 @@
+	R_ICU->IELSR_b[VECTOR_NUMBER_CAN_GLERR].IELS = ELC_EVENT_CAN_GLERR;                        \
+	R_ICU->IELSR_b[VECTOR_NUMBER_CAN_RXF].IELS = ELC_EVENT_CAN_RXF;                            \
+	IRQ_CONNECT(VECTOR_NUMBER_CAN_GLERR,                                                       \
+		    DT_IRQ_BY_NAME(DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), glerr,  \
+				   priority),                                                      \
+		    canfd_error_isr, NULL, 0);                                                     \
+	IRQ_CONNECT(VECTOR_NUMBER_CAN_RXF,                                                         \
+		    DT_IRQ_BY_NAME(DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), rxf,    \
+				   priority),                                                      \
+		    canfd_rx_fifo_isr, NULL, 0);                                                   \
+	irq_enable(VECTOR_NUMBER_CAN_RXF);                                                         \
+	irq_enable(VECTOR_NUMBER_CAN_GLERR);
+
+static canfd_global_cfg_t g_canfd_global_cfg = {
+	.global_interrupts = CANFD_CFG_GLERR_IRQ,
+	.global_config = CANFD_CFG_GLOBAL,
+	.rx_mb_config = CANFD_CFG_RXMB,
+	.global_err_ipl = DT_IRQ_BY_NAME(DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global),
+					 glerr, priority),
+	.rx_fifo_ipl = DT_IRQ_BY_NAME(DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), rxf,
+				      priority),
+	.rx_fifo_config = CANFD_CFG_RXFIFO,
+	.common_fifo_config = CANFD_CFG_COMMONFIFO0,
+};
+
+static const struct can_renesas_ra_global_cfg g_can_renesas_ra_global_cfg = {
+	.op_clk = DEVICE_DT_GET(DT_CLOCKS_CTLR_BY_NAME(
+		DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), opclk)),
+	.ram_clk = DEVICE_DT_GET(DT_CLOCKS_CTLR_BY_NAME(
+		DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), ramclk)),
+	.op_subsys = {.mstp = (uint32_t *)DT_CLOCKS_CELL_BY_NAME(
+			      DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), opclk, mstp),
+		      .stop_bit = DT_CLOCKS_CELL_BY_NAME(
+			      DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), opclk,
+			      stop_bit)},
+	.ram_subsys = {.mstp = (uint32_t *)DT_CLOCKS_CELL_BY_NAME(
+			       DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), ramclk,
+			       mstp),
+		       .stop_bit = DT_CLOCKS_CELL_BY_NAME(
+			       DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), ramclk,
+			       stop_bit)},
+};
+
+static int can_renesas_ra_global_init(const struct device *dev)
+{
+	int ret;
+	const struct can_renesas_ra_global_cfg *cfg = dev->config;
+
+	ret = clock_control_on(cfg->op_clk, (clock_control_subsys_t)&cfg->op_subsys);
+	ret |= clock_control_on(cfg->ram_clk, (clock_control_subsys_t)&cfg->ram_subsys);
+	if (ret) {
+		LOG_DBG("driver: can: init: clock initialize failed");
+		return ret;
+	}
+	CAN_RENESAS_RA_GLOBAL_IRQ_INIT();
+
+	return 0;
+}
+
+DEVICE_DT_DEFINE(DT_COMPAT_GET_ANY_STATUS_OKAY(renesas_ra_canfd_global), can_renesas_ra_global_init,
```

Comment:
Is the `global` always a singleton?

Reply:
Yes. All CAN controller nodes will share 1 global node as their parent.